server/routes/songs.js

```diff
@@ -0,0 +1,80 @@
+var express = require("express");
+var router = express.Router();
+const { search_songs, generate_recommendations } = require("../api");
+
+/* GET song search results . */
+router.get("/search/:searchString", async function (req, res, next) {
+  const resp = await search_songs(req.params.searchString);
+  return res.json(resp);
+});
+
+// GET generate song recommendations (maybe make this a PUT to cache the recommendations?)
+router.get(
+  "/recommendations/generate/:songId",
+  async function (req, res, next) {
+    const resp = await generate_recommendations(req.params.songId, true); // todo - change to false when ML model is implemented
+    return res.json(resp);
+  }
+);
+
+// todo
+// GET scoreboard data
+router.get("/scoreboard", function (req, res, next) {
+  // call database
+  const scoreboard = { scoreboard: "scoreboard" };
+  return res.json(scoreboard);
+});
+
+// todo
+// GET globally searched songs
+router.get("/globallysearched", function (req, res, next) {
+  // call database
+  const globallysearched = { globallysearched: "globallysearched" };
+  return res.json(globallysearched);
+});
+
+// todo
+// GET song liked status
+// Helpful - https://github.com/fingerprintjs/fingerprintjs for generating user session id on client side
```

Comment:
the fingerprint library might be a little excessive for our use, but (i think you mentioned this on tuesday) i guess it's our workaround for not having auth to avoid voting abuse?